### PR TITLE
No more coroutines

### PR DESCRIPTION
**Pull request by blampe**
These are never a good idea at this layer, let's get in front of the problem.

@breerly @abhinav

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,10 @@ clean:
 test: clean
 	$(pytest) $(test_args)
 
+.PHONE: lint
+lint:
+	flake8 yarpc tests examples setup.py
+	! grep -r coroutine yarpc
```
Comment (abhinav) [inline Makefile]: nice

Comment (breerly) [inline Makefile]: neato

**Comment by breerly** [comment]
looks like this breaks exception handling?

**Comment by blampe** [comment]
Yeah we could copy @abhinav's `fail_to` into this. @breerly mind if I punt that for a separate diff?

**Comment by breerly** [comment]
Sure why not :shipit:

**Comment by breerly** [comment]
Nice to have no coroutines from the gitgo

**Comment by abhinav** [comment]
:+1:

**Pull request merged**